packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -139,6 +145,9 @@ function DashboardLayout(props: DashboardLayoutProps) {
   const navigationContext = React.useContext(NavigationContext);
   const appWindowContext = React.useContext(WindowContext);
 
+  const branding = { ...brandingContext, ...brandingProp };
+  const navigation = navigationProp ?? navigationContext;
```

Comment:
I think it's important to document that the `navigation` override in the `DashboardLayout` completely replaces the one passed in the parent `AppProvider`, whereas the `branding` prop is merged with the parent `branding`